File: Project2/src/main/java/com/project/service/CommentService.java
package com.project.service;

import org.springframework.beans.factory.annotation.Autowired;

import com.project.dao.CommentDao;
import com.project.model.Comment;

public interface CommentService {

	
	public Comment getCommentById(int commentId);
	
	public long newComment(int userId, int postId, String text);
	public long updateComment(int commentId, String text);
	
	public long deleteComment(int commentId);
	
}

File: Project2/src/main/webapp/Angular/src/app/shared/comment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '../../../node_modules/@angular/common/http';
import { CommentData } from '../models/commentdata.class';
import { map } from '../../../node_modules/rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class CommentService {

  constructor(private client: HttpClient) { }

  newComment(comment: CommentData) {
    const userId = '' + comment.user.userId;
    const postId = '' + comment.post;
    const text = comment.text;
    const url = 'http://localhost:12345/Project2/newComment.do';
    return this.client.post(url, null, {
      params: {
        userId,
        postId,
        text
      }
    }).pipe(map(resp => resp as number));
  }

  deleteComment(comment: CommentData) {
    const commentId = comment.commentId + '';
    const url = 'http://localhost:12345/Project2/deleteComment.do?commentId=' + commentId;
    return this.client.get(url);
  }
}
